### PR TITLE
ColorFilterArray: fix negative shift UB (coverty report)

### PR DESCRIPTION
**Pull request by axxel**
Also add test for negative y shift.

### CODE REVIEW

File: src/librawspeed/metadata/ColorFilterArrayTest.cpp [hunk 1]
```diff
@@ -135,6 +135,7 @@ TEST_P(ColorFilterArrayTest, DcrawFilterShift1) {
     ASSERT_EQ(ColorFilterArray::shiftDcrawFilter(rggb, 0, 1), gbrg);
     ASSERT_EQ(ColorFilterArray::shiftDcrawFilter(rggb, 1, 1), bggr);
     ASSERT_EQ(ColorFilterArray::shiftDcrawFilter(rggb, 0, 2), rggb);
+    ASSERT_EQ(ColorFilterArray::shiftDcrawFilter(rggb, 0, -1), gbrg);
```
Comment (LebedevRI) [inline src/librawspeed/metadata/ColorFilterArrayTest.cpp:138]: JFYI, this test executes all the same shiftDcrawFilter checks 255 times.
But i'l rewrite that locally.

Comment (axxel) [reply src/librawspeed/metadata/ColorFilterArrayTest.cpp:138]: I thought so but did not have the courage to look into the details of the google test framework to fix that. Thanks for taking care of that.

**Comment by LebedevRI** [comment]
Thank you for fixing `shiftLeft()` / `shiftDown()` to work with negative offsets :)

File: src/librawspeed/metadata/ColorFilterArrayTest.cpp (hunk 1 above)
Comment (LebedevRI) [reply src/librawspeed/metadata/ColorFilterArrayTest.cpp:138]: There, now this executable will run 6744 tests :)
